assets/src/blocks/widgets/event-attendees/edit.js

```diff
@@ -0,0 +1,263 @@
+/**
+ * WordPress dependencies
+ */
+import { InspectorControls } from '@wordpress/editor';
+import { Component, Fragment } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { PanelBody, Placeholder, Spinner, ToggleControl } from '@wordpress/components';
+
+/**
+ * External dependencies
+ */
+import { __ } from '@eventespresso/i18n';
+import {
+	EditorDatetimeSelect,
+	EditorEventSelect,
+	EditorStatusSelect,
+	EditorTicketSelect,
+	QueryLimit,
+} from '@eventespresso/components';
+import * as statusModel from '../../../data/model/status';
+import PropTypes from 'prop-types';
+import { isEmpty, uniqBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { EventAttendeesList } from './list';
+
+class EventAttendeesEditor extends Component {
+	static propTypes = {
+		attendees: PropTypes.array,
+		isLoading: PropTypes.bool,
+		attributes: PropTypes.shape( {
+			eventId: PropTypes.number,
+			datetimeId: PropTypes.number,
+			ticketId: PropTypes.number,
+			status: PropTypes.string,
+			showGravatar: PropTypes.bool,
+			displayOnArchives: PropTypes.bool,
+			limit: PropTypes.number,
+		} ),
+	};
+
+	static defaultProps = {
+		attendees: [],
+		isLoading: true,
+		attributes: {
+			eventId: 0,
+			datetimeId: 0,
+			ticketId: 0,
+			status: 'RAP',
+			showGravatar: true,
+			displayOnArchives: false,
+			limit: -1
+		},
+	};
+
+	setEventId = ( eventId ) => {
+		const value = eventId !== null && eventId.value ?
+			parseInt( eventId.value, 10 ) :
+			0;
+		this.props.setAttributes(
+			{
+				eventId: value,
+				datetimeId: 0,
+				ticketId: 0,
+			}
+		);
+	};
+
+	setDatetimeId = ( datetimeId ) => {
+		const value = datetimeId !== null && datetimeId.value ?
+			parseInt( datetimeId.value, 10 ) :
+			0;
+		this.props.setAttributes(
+			{
+				datetimeId: value,
+				ticketId: 0,
+			}
+		);
+	};
+
+	setTicketId = ( ticketId ) => {
+		const value = ticketId !== null && ticketId.value ?
+			parseInt( ticketId.value, 10 ) :
+			0;
+		this.props.setAttributes( { ticketId: value } );
+	};
+
+	setStatus = ( status ) => {
+		const value = status !== null && status.value ? status.value : 'ANY';
```

Comment:
I added that in order to deal with how the defaults were being set in the PHP version of the shortcode.

Basically the issue is that if someone is using `[ESPRESSO_EVENT_ATTENDEES]` in a post and does not specify a post status, then the shortcode defaults to using `RAP` so that only attendees for approved registrations are displayed. With the react component however, it is possible to deselect the reg status and essentially display all attendees regardless of their reg status. This unfortunately sets the status value in the block attributes to `null` which when passed to the PHP code looks exactly like a shortcode that has just not set the reg status and should therefore default to `RAP`.

So in short we have the following cases

| attribute source | status attribute value | should default to |
| --- | --- | --- |
| shortcode in post content | `RAP` | `RAP` |
| shortcode in post content | `null` | `RAP` |
| gutenberg block | `RAP` | `RAP` |
| gutenberg block | `null` | no reg status filtering |

On further thought... we may want to remove the ability to deselect the reg status in the Gutenberg block because it is highly unlikely that admins would really want to display a list of attendees that includes registrations that have been cancelled or declined. The downside to this is that it does not allow lists to show both Approved and Pending Payment and even Wait List which someone may be inclined to want to do.

Reply:
I think this is an opportunity to improve things so that the block is better than the shortcode (better meaning the ability to not select a status).  In my opinion any back-compat behaviour to preserve from the shortcode should just be handled by the shortcode transform property.  Rendering server side should be modified for this _blocks_ so that it now handles when there is no reg status set.  Either that, or preserve behaviour with the shortcode and the selector always defaults to RAP with the disabling of clearing selections (which is doable).